x-pack/plugins/apm/server/lib/transactions/get_anomaly_data/index.ts

```diff
@@ -60,48 +53,79 @@ export async function getAnomalySeries({
     .some((uiFilterName) => !knownFilters.includes(uiFilterName));
 
   if (hasUnknownFiltersApplied) {
-    return;
+    return undefined;
   }
 
-  // don't fetch anomalies if the ML plugin is not setup
-  if (!setup.ml) {
-    return;
-  }
+  const { intervalString } = getBucketSize({ start, end });
 
-  // don't fetch anomalies if required license is not satisfied
-  const mlCapabilities = await setup.ml.mlSystem.mlCapabilities();
-  if (!mlCapabilities.isPlatinumOrTrialLicense) {
-    return;
-  }
+  // move the start back with one bucket size, to ensure to get anomaly data in the beginning
+  // this is required because ML has a minimum bucket size (default is 900s) so if our buckets
+  // are smaller, we might have several null buckets in the beginning
+  const mlStart = start - 900 * 1000;
 
-  const mlJobIds = await getMLJobIds(setup.ml.anomalyDetectors, environment);
+  const [anomaliesResponse, jobIds] = await Promise.all([
+    anomalySeriesFetcher({
+      serviceName,
+      transactionType,
+      intervalString,
+      ml,
+      start: mlStart,
+      end,
+    }),
+    getMLJobIds(ml.anomalyDetectors, environment),
+  ]);
 
-  const jobId = mlJobIds[0];
+  const scoreSeriesCollection = anomaliesResponse?.aggregations?.job_id.buckets
+    .filter((bucket) => jobIds.includes(bucket.key as string))
+    .map((bucket) => {
+      const dateBuckets = bucket.ml_avg_response_times.buckets;
 
-  const mlBucketSize = await getMlBucketSize({ setup, jobId, logger });
-  if (!isNumber(mlBucketSize)) {
-    return;
-  }
+      return {
+        jobId: bucket.key as string,
+        anomalyScore: compact(
+          dateBuckets.map((dateBucket) => {
+            const metrics = maybe(dateBucket.anomaly_score.top[0])?.metrics;
+            const score = metrics?.record_score ?? null;
+
+            if (
+              !metrics ||
+              score === null ||
+              score < ANOMALY_THRESHOLD.CRITICAL
```

Comment:
nit: do we need to explicitly check whether `score` is `null` when we also check that it is less than 75 (critical)?
```suggestion
              score < ANOMALY_THRESHOLD.CRITICAL
```

Reply:
TS will complain when using arithmetic operators on maybe-null values: https://www.typescriptlang.org/play?#code/MYewdgzgLgBBoCcCmAuMBXAtgIyQmAPjBgDYkwC8x6ZA3AFCMCWAZgBTwjIwA8MA7AFYAlDADe9GFJjIo6BGBhQE6JAwC+QA

Reply:
Another approach is to check for both `null` and `undefined`. It seems a little brittle to rely on `score` being specifically `null` or a number.

```suggestion
              score == null ||
              score < ANOMALY_THRESHOLD.CRITICAL
```

If we do above change we can remove the default value:

```diff
-const score = metrics?.record_score ?? null;
+const score = metrics?.record_score;

Reply:
Yet another approach:
```suggestion
              !Number.isInteger(score) || 
              score < ANOMALY_THRESHOLD.CRITICAL
```

Again, just trying to make this explicit so someone doesn't change this inadvertently an breaks it down the line

Reply:
👍 I'll use `isFiniteNumber` to refine.